### DISCUSSION TITLE
best practices

### DISCUSSION THREAD

**Discussion by Manubi** [tauri-apps/tauri#8338] [Q&A]
Hi, 
I'd like to ask about your experience with different frontend frameworks and tauri? What do you think is the frontend framework that makes the least trouble when developing? I know it comes down to how familiar you are with the tech, but I just want to have a general feel what you guys think has at the moment the best developer experience? 
Thanks!! ❤️

**Comment by FabianLars** [comment]
I'm not a frontend guy myself but i talked with _many_ devs using tauri and at least if you're not a pro with the chosen frontend stack people tend to have many more issues with frameworks that offer an SSR mode, especially since they tend to default to that.

SSR does _not_ work in Tauri apps and a non-SSR mode in frameworks that fall into the described category often feels like a second (or third) class citizen if that makes sense?

For that reason i pretty much always recommend vite plus the "basic" frontend framework (react, vue, solid, svelte, etc...) instead of their meta framework counterpart (next, nuxt, solidstart, sveltekit) unless you can't live without a file based router no matter what.

**Reply by Manubi** [reply]
Thanks for the reply. Kinda what I expected. Apart from client site rendering doesn't really matter if i choose svelte or react? :)

**Reply by FabianLars** [reply]
yep, at this point it's really only about personal preference imo

**Comment by lutfi-haslab** [comment]
I just read this forum after setting up my tauri with NextJS. ugh, yes, we should'nt using meta framework that enable SSR on first priority. Is sidecar nodejs has support for Mobile?

**Reply by FabianLars** [reply]
I don't think it works on mobile, at least not yet.
